### ISSUE TITLE
False negative `Lint/DuplicateMethods`

### ISSUE THREAD

**Issue by Nittarab** [rubocop/rubocop#10182]
*Be clear, concise and precise in your description of the problem.
Open an issue with a descriptive title and a summary in grammatically correct,
complete sentences.*

*Use the template below when reporting bugs. Please, make sure that
you're running the latest stable RuboCop and that the problem you're reporting
hasn't been reported (and potentially fixed) already.*

*Before filing the ticket you should replace all text above the horizontal
rule with your own words.*

--------

## Expected behavior

There shouldn't be any warning.

## Actual behavior

```
lib/mymodule/configuration.rb:16:5: W: Lint/DuplicateMethods: Method MyModule::Configuration#credentials is defined at both lib/MyModule/configuration.rb:7 and lib/mymodule/configuration.rb:16.
def credentials(email = '', password = '', strategy: 'local')
```
## Steps to reproduce the problem

```yaml
 Lint/DuplicateMethods:
  Enabled: true
```

```ruby
module MyModule
  class Configuration
    Credentials = Struct.new(:client_id, :client_secret, :strategy)

    attr_accessor :api_endpoint, :api_version, :proxy_host, :proxy_port, :locale, :strategy, :credentials

    def initialize
      @api_endpoint = URI('https://tet.com/')
      @api_version = 'v1'
      @locale = I18n.locale
      @strategy = 'local'
    end

    def credentials(email = '', password = '', strategy: 'local')
      return @credentials if @credentials

      @credentials = Credentials.new(email, password, strategy)
    end
  end
end
```

## RuboCop version

`1.22.1 (using Parser 3.0.2.0, rubocop-ast 1.12.0, running on ruby 3.0.2 x86_64-darwin20)`
